chaosLib/litmus/aws_az_chaos/lib/aws_az_chaos.py

```diff
@@ -0,0 +1,150 @@
+
+import pkg.types.types  as types
+import pkg.events.events as events
+import logging
+import pkg.utils.common.common as common
+import pkg.utils.exec.exec as litmusexec
+import pkg.types.types  as types
+import pkg.events.events as events
+import logging
+import pkg.utils.common.common as common
+from datetime import datetime
+import pkg.maths.maths as maths
+
+#PrepareAWSAZExperiment contains the prepration steps before chaos injection
+def PrepareAWSAZExperiment(experimentsDetails , resultDetails, eventsDetails, chaosDetails, clients, statusAws):
+
+	# Waiting for the ramp time before chaos injection
+	if experimentsDetails.RampTime != 0 :
+		logging.info("[Ramp]: Waiting for the %s ramp time before injecting chaos",experimentsDetails.RampTime)
+		common.WaitForDuration(experimentsDetails.RampTime)
+	
+ 	# mode for chaos injection
+	if experimentsDetails.Sequence.lower() == "serial":
+		err = injectChaosInSerialMode(experimentsDetails, chaosDetails, eventsDetails, resultDetails, clients, statusAws)
+		if err != None:
+			return err
+	elif experimentsDetails.Sequence.lower() == "parallel":
+		err = injectChaosInParallelMode(experimentsDetails, chaosDetails, eventsDetails, resultDetails, clients, statusAws)
+		if err != None:
+			return err
+	else:
+		return ValueError("{} sequence is not supported".format(experimentsDetails.Sequence))
+
+	# Waiting for the ramp time after chaos injection
+	if experimentsDetails.RampTime != 0 :
+		logging.info("[Ramp]: Waiting for the %s ramp time after injecting chaos",experimentsDetails.RampTime)
+		common.WaitForDuration(experimentsDetails.RampTime)
+
+	return None
+
+# injectChaosInSerialMode disable the target available zone from loadbalancer in serial mode(one by one)
+def injectChaosInSerialMode(experimentsDetails , chaosDetails , eventsDetails , resultDetails, clients, statusAws): 
+	
+	#ChaosStartTimeStamp contains the start timestamp, when the chaos injection begin
+	ChaosStartTimeStamp = datetime.now()
+	duration = (datetime.now() - ChaosStartTimeStamp).seconds
+	
+	while duration < experimentsDetails.ChaosDuration:
+     
+		# Get the target available zones for the chaos execution
+		targetZones = experimentsDetails.LoadBalancerZones.split(",")
+		
+		logging.info("[Info]: Target available zone list, %s", targetZones)
+
+		if experimentsDetails.EngineName != "" :
+			msg = "Injecting " + experimentsDetails.ExperimentName + " chaos on available zone"
+			types.SetEngineEventAttributes(eventsDetails, types.ChaosInject, msg, "Normal", chaosDetails)
+			events.GenerateEvents(eventsDetails, chaosDetails, "ChaosEngine", clients)
+		
+		# Detaching the target zones from loa balancer 
+		for azone in targetZones:
+
+			logging.info("[Info]: Detaching the following zone(s), Zone Name : %s", azone)
```

Comment:
```suggestion
			logging.info("[Info]: Detaching the following zone, Zone Name : %s", azone)
``

Reply:
It can be multiple or single so that is why we are adding `(s)`

Reply:
Since we are looping through a list of comma-separated zones, I guess in every iteration `azone` will consist of only one zone name.